packages/clay-alert/src/ClayAlert.tsx

```diff
@@ -0,0 +1,177 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+import Icon from '@clayui/icon';
+
+const useAutoClose = (autoClose?: boolean | number, onClose = () => {}) => {
+	const timer = React.useRef<number | undefined>(undefined);
```

Comment:
hey @bryceosterhaus just out of curiosity why are we storing the timer inside a `useRef`?

Reply:
This is sort of a paradigm shift that comes from classes to hooks. Because call components are just functions now, we need a way to keep a value around. I got the idea from other examples I saw that used `setTimeout`. I then read this in the docs and helped me make sense of it. https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables

Reply:
Hm, thanks for the information, it seems clearer now the use of `useRef` I had another thought about it.